reconcile/terraform_resources.py

```diff
@@ -549,17 +549,40 @@ def init_working_dirs(
     return ts, working_dirs
 
 
+def filter_accounts_by_name(
+    accounts: Iterable[Mapping[str, Any]], filter: Iterable[str]
+) -> Collection[Mapping[str, Any]]:
+    return [ac for ac in accounts if ac["name"] in filter]
+
+
+def exclude_accounts_by_name(
+    accounts: Iterable[Mapping[str, Any]], filter: Iterable[str]
+) -> Collection[Mapping[str, Any]]:
+    return [ac for ac in accounts if ac["name"] not in filter]
+
+
 def setup(
     dry_run: bool,
     print_to_file: str,
     thread_pool_size: int,
     internal: str,
     use_jump_host: bool,
     account_names: Optional[Collection[str]],
+    exclude_accounts: Optional[Collection[str]],
 ) -> tuple[ResourceInventory, OC_Map, Terraform, ExternalResourceSpecInventory]:
     accounts = queries.get_aws_accounts(terraform_state=True)
+    if not account_names and exclude_accounts:
+        excluding = filter_accounts_by_name(accounts, exclude_accounts)
+        if len(excluding) != len(exclude_accounts):
+            raise ValueError(
+                f"Accounts {exclude_accounts} were provided as arguments, but not found in app-interface. Check your input for typos or for missing AWS account definitions."
+            )
+        accounts = exclude_accounts_by_name(accounts, exclude_accounts)
+        if len(accounts) == 0:
+            raise ValueError("You have excluded all aws accounts, verify your input")
+        account_names = tuple(ac["name"] for ac in accounts)
     if account_names:
-        accounts = [n for n in accounts if n["name"] in account_names]
+        accounts = filter_accounts_by_name(accounts, account_names)
```

Comment:
`account_names` is function parameter, should avoid mutate or reassign it, can rename it in parameter to `include_accounts` to make it consistent with `exclude_accounts`

Validate account exist by name logic is duplicated in both branches, can be extracted to a function.

So this can be like:

```python
def validate_account_names(
    accounts: Iterable[Mapping[str, Any]], names: Iterable[str]
):
    if len(accounts) != len(names):
        missing_names = set(names) - {a["name"] for a in accounts}
        raise ValueError(
            f"Accounts {missing_names} were provided as arguments, but not found in app-interface. Check your input for typos or for missing AWS account definitions."
        )

def setup(
    dry_run: bool,
    print_to_file: str,
    thread_pool_size: int,
    internal: str,
    use_jump_host: bool,
    include_accounts: Optional[Collection[str]],
    exclude_accounts: Optional[Collection[str]],
) -> tuple[ResourceInventory, OC_Map, Terraform, ExternalResourceSpecInventory]:
    accounts = queries.get_aws_accounts(terraform_state=True)
    if not include_accounts and exclude_accounts:
        excluding = filter_accounts_by_name(accounts, exclude_accounts)
        validate_account_names(excluding, exclude_accounts)
        accounts = exclude_accounts_by_name(accounts, exclude_accounts)
        if len(accounts) == 0:
            raise ValueError("You have excluded all aws accounts, verify your input")
    elif include_accounts:
        accounts = filter_accounts_by_name(accounts, include_accounts)
        validate_account_names(accounts, include_accounts)
    account_names = [a["name"] for a in accounts]
```
Can even extract the whole get and filter/exclude accounts logic to a function if this becomes too complex.

May need to update exception string in tests as it was using `tuple` in `ValueError` before.

Reply:
Thank you for the review.

Reply:
I've implemented your suggestions, thanks again.